udata/core/badges/tasks.py

```diff
@@ -23,20 +23,57 @@ def call_task(badge, **kwargs):
     return wrapper
 
 
-def owner_recipients(badge):
-    if badge.subject.organization:
-        members = badge.subject.organization.members
-    else:
-        members = badge.subject.members
-    return [m.user for m in members]
+@connect(on_badge_added)
+def notify_badge_added_certified(sender, the_badge=None):
+    '''
+    Send an email when a `CERTIFIED` badge is added to an `Organization`
+
+    Parameters
+    ----------
+    sender
+        The object that emitted the event.
+    the_badge
+        The `Badge` object awarded.
+        NB: `badge` is a reserved param name in Blinker.
+    '''
+    if the_badge.kind == CERTIFIED and isinstance(sender, Organization):
+        recipients = [member.user for member in sender.members]
+        subject = _(
+            'Your organization "%(name)s" has been certified',
+            name=sender.name
+        )
+        mail.send(
+            subject,
+            recipients,
+            'badge_added_certified',
+            organization=sender,
+            badge=the_badge
+        )
 
 
 @connect(on_badge_added)
-def notify_badge_added(badge):
-    if isinstance(badge.subject, Organization):
-        recipients = owner_recipients(badge)
-        subject = _('Your %(type)s gain a new badge',
-                    type=badge.subject.verbose_name)
-        mail.send(subject, recipients, 'badge_added', badge=badge)
-    else:
-        log.warning('Unrecognized badge subject type %s', type(badge.subject))
+def notify_badge_added_public_service(sender, the_badge=None):
+    '''
+    Send an email when a `PUBLIC_SERVICE` badge is added to an `Organization`
+
+    Parameters
+    ----------
+    sender
+        The object that emitted the event.
+    the_badge
+        The `Badge` object awarded.
+        NB: `badge` is a reserved param name in Blinker.
+    '''
+    if the_badge.kind == PUBLIC_SERVICE and isinstance(sender, Organization):
+        recipients = [member.user for member in sender.members]
+        subject = _(
+            'Your organization "%(name)s" has been identified as public service',
```

Comment:
The subject does not contain `a public service` as in emails body. Not sure which one is correct.